File: admin-backend/admin-system/src/main/java/io/github/faustofan/admin/system/domain/model/SysTenant.java
package io.github.faustofan.admin.system.domain.model;

import io.github.faustofan.admin.shared.persistence.model.BaseEntity;
import io.github.faustofan.admin.system.domain.enums.TenantStatus;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.Key;
import org.babyfish.jimmer.sql.*;
import java.time.LocalDateTime;

/**
 * 系统租户实体
 * 表示系统中的租户信息
 */
@Entity
@Table(name = "sys_tenant")
public interface SysTenant extends BaseEntity {

    @Key
    String tenantCode();

    String name();

    // 租户关联一个产品包
    @ManyToOne
    @JoinColumn(name = "package_id")
    SysProductPackage packageInfo();

    LocalDateTime expireTime();

    // 修改为枚举类型
    @Column(name = "status")
    // 可以设置默认值，insert时如果未赋值则使用此值
    @Default("ENABLE")
    TenantStatus status();
}

File: src/main/kotlin/io/github/faustofan/admin/system/domain/model/SysTenant.kt
package io.github.faustofan.admin.system.domain.model

import io.github.faustofan.admin.shared.domain.model.BaseEntity
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.EnumType
import org.babyfish.jimmer.sql.Key
import java.time.LocalDateTime

@Entity
interface SysTenant : BaseEntity {

    // 租户名称，唯一键
    @Key
    val name: String

    // 租户编码 (用于二级域名或系统标识)
    @Key
    val code: String

    // 联系人
    val contactName: String?

    // 状态: ENABLE, DISABLE
    val status: TenantStatus

    // 过期时间
    val expireTime: LocalDateTime?
}

@EnumType
enum class TenantStatus {
    ENABLE, DISABLE
}